personalrobotics/ada_ros2
# Configure `ada_moveit` for Articulable Fork

# Description

This pull request modifies launch files to adapt to the specified `end_effector_tool`. This will allow users to specify tools from `fork`, and `articulable_fork`. For developers using ADA before the articulable fork hardware is transitioned in, set the tool to `fork`.

A prerequisite to testing these changes is having your development machine setup according to the [Software Installation](https://docs.google.com/document/d/1-fPd2vl-8c38OCNbtztFirrAGxInyE0txhr0YHg-0zI/edit?tab=t.0) section of the Articulable Fork Assembly Guide

Related PRs: https://github.com/personalrobotics/ada_ros2/pull/52, https://github.com/personalrobotics/ada_ros2/pull/53, https://github.com/personalrobotics/ada_feeding/pull/211

# Testing procedure

- \[x\] To test in simulation, run `ros2 launch ada_moveit demo.launch.py sim:=mock end_effector_tool:=<tool>` for each `<tool>` in `fork`, `articulable_fork`
  - \[x\] In `RViz2`, observe that the appropriate robot description has been loaded visually
  - \[x\] Within the `MotionPlanning` window in `RViz2`, open the `PlanningGroup` drop-down menu and observe that the available options are `ada`, `articulable_fork`, `hand`, and `jaco_arm`.
  - \[x\] Using the `ada` planning group, set the goal state to random valid, then plan and execute the trajectory and verify that ADA moves to the desired configuration
- \[x\] To test on the real hardware, first connect the appropriate hardware to your development machine via USB, e.g. for `end_effector_tool = fork` only connect the Jaco to your machine, and for `end_effector_tool = articulable_fork` connect both the Jaco and Articulable Fork to your machine
  - \[x\] Open a new terminal and run `ros2 run forque_sensor_hardware forque_sensor_hardware --ros-args -p host:=ft-sensor-2`
  - \[x\] Open a new terminal and run `ros2 launch ada_moveit demo.launch.py sim:=mock end_effector_tool:=<tool>` for each `<tool>` in `fork`, `articulable_fork`
  - \[x\] Activate the necessary controllers (`jaco_arm_controller`, `af_controller`) by running 
  - \[x\] Using the `ada` planning group, adjust the position of the end-effector frame such that all joints of interest will be actuated without colliding with the test environment, then plan and execute the trajectory and verify that ADA moves to the desired configuration

# Before opening a pull request

- \[x\] `pre-commit run --all-files`
- \[x\] Run your code through [pylint](https://pylint.readthedocs.io/en/latest/). `pylint --recursive=y --rcfile=.pylintrc .`. All warnings but `fixme` must be addressed.

# Before Merging

- \[ \] `Squash & Merge`
